xsd/types.go

```diff
@@ -221,122 +221,11 @@ func (t Type) IsBinary() bool {
 	}
 }
 
-// ToGoType maps the XSD type to the appropriate Go type.
-func (t Type) ToGoType() string {
-	switch t {
-	// String types
-	case String, NormalizedString, Token, Language, NMTOKEN, Name, NCName,
-		ID, IDREF, ENTITY, AnyURI:
-		return "string"
-
-	// String slice types
-	case NMTOKENS, IDREFS, ENTITIES:
-		return "[]string"
-
-	// Boolean type
-	case Boolean:
-		return "bool"
-
-	// Floating point types
-	case Float:
-		return "float32"
-	case Double, Decimal:
-		return "float64"
```

Comment:
Was this the culprit?was this the culprit? 👀

Reply:
yeah, the culprit was using `float32` to parse the data, and then upcasting it to float64 - upcasting to float64 is what caused the extra decimals to be added (due to how casting is handled) better to parse straight into a float64 and avoid the casting